crates/bevy_text/src/pipeline.rs

```diff
@@ -114,4 +114,118 @@ impl TextPipeline {
+
+                let section = TextMeasureSection {
+                    font_id: FontId(i),
+                    scale: PxScale::from(font_size),
+                    text: section.value.clone(),
+                };
+
+                Ok(section)
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+
+        let mut info = TextMeasureInfo {
+            fonts: auto_fonts,
+            scaled_fonts,
+            sections,
+            text_alignment,
+            linebreak_behaviour: linebreak_behaviour.into(),
+            min_width_content_size: Vec2::ZERO,
+            max_width_content_size: Vec2::ZERO,
+        };
+
+        let section_texts = info.prepare_section_texts();
+        let min = info.compute_size_from_section_texts(&section_texts, Vec2::new(0.0, f32::INFINITY));
+        let max = info.compute_size_from_section_texts(&section_texts, Vec2::new(f32::INFINITY, f32::INFINITY));
+        info.min_width_content_size = min;
+        info.max_width_content_size = max;
+
+        Ok(info)
+    }
+}
+
+#[derive(Clone)]
+pub struct TextMeasureSection {
+    pub text: String,
+    pub scale: PxScale,
+    pub font_id: FontId,
+}
+
+#[derive(Clone)]
+pub struct TextMeasureInfo {
+    pub fonts: Vec<ab_glyph::FontArc>,
+    pub scaled_fonts: Vec<ab_glyph::PxScaleFont<ab_glyph::FontArc>>,
+    pub sections: Vec<TextMeasureSection>,
+    pub text_alignment: TextAlignment,
+    pub linebreak_behaviour: glyph_brush_layout::BuiltInLineBreaker,
+    pub min_width_content_size: Vec2,
+    pub max_width_content_size: Vec2,
+}
+
+impl TextMeasureInfo {
+    fn prepare_section_texts(&self) -> Vec<SectionText> {
+        self.sections
+            .iter()
+            .map(|section| SectionText {
+                font_id: section.font_id,
+                scale: section.scale,
+                text: &section.text,
+            })
+            .collect::<Vec<_>>()
+    }
+
+    fn compute_size_from_section_texts(&self, sections: &[SectionText], bounds: Vec2) -> Vec2 {
+        let geom = SectionGeometry {
+            bounds: (bounds.x, bounds.y),
+            ..Default::default()
+        };
+        let section_glyphs = glyph_brush_layout::Layout::default()
+            .h_align(self.text_alignment.into())
+            .line_breaker(self.linebreak_behaviour)
+            .calculate_glyphs(&self.fonts, &geom, &sections);
+
+        let mut min_x: f32 = std::f32::MAX;
+        let mut min_y: f32 = std::f32::MAX;
+        let mut max_x: f32 = std::f32::MIN;
+        let mut max_y: f32 = std::f32::MIN;
+
+        for sg in section_glyphs {
+            let scaled_font = &self.scaled_fonts[sg.section_index];
+            let glyph = &sg.glyph;
+            min_x = min_x.min(glyph.position.x);
+            min_y = min_y.min(glyph.position.y - scaled_font.ascent());
+            max_x = max_x.max(glyph.position.x + scaled_font.h_advance(glyph.id));
+            max_y = max_y.max(glyph.position.y - scaled_font.descent());
```

Comment:
I'm not really familiar with the font logic, but is the `-` correct here?
Looks suspicious because the `max` is calculated.

Reply:
Yes I think so, fonts use a coordinate system increasing upwards so ascent is a positive value and descent is negative, but Bevy UI uses a downwards increasing coordinate system, so we have to subtract from the baseline position to get the minimum and maximum values.

We only care about the size here though, not direction, maybe this can be simplified somehow.